genericfilestorage/storage_test.go

```diff
@@ -0,0 +1,104 @@
+package genericfilestorage
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_givenBulkDownloadFail_whenDownloadFilesCalled_thenExpectError(t *testing.T) {
+	// Given
+	files := givenFiles()
+	mockFileProvider := givenMockFileProvider().
+		GivenGetFilesSucceed(files)
+	mockBulkdownloader := givenMockBulkFileDownloader().
+		GivenDownloadFilesFail(errors.New("sad error"))
+	storage := Storage{
+		bulkdownloader: mockBulkdownloader,
+		fileprovider:   mockFileProvider,
+		envKey:         "whatever",
+	}
+
+	// When
+	path, err := storage.DownloadFiles()
+
+	// Then
+	assert.Error(t, err)
+	assert.Empty(t, path)
+}
+
+func Test_givenBulkDownloadSucceeds_whenDownloadFilesCalled_thenExpectNoError(t *testing.T) {
+	// Given
+	files := givenFiles()
+	mockFileProvider := givenMockFileProvider().
+		GivenGetFilesSucceed(files)
+	mockBulkdownloader := givenMockBulkFileDownloader().
+		GivenDownloadFilesSucceed()
+	storage := Storage{
+		bulkdownloader: mockBulkdownloader,
+		fileprovider:   mockFileProvider,
+		envKey:         "whatever",
+	}
+
+	// When
+	path, err := storage.DownloadFiles()
+
+	// Then
+	assert.NoError(t, err)
+	assert.NotEmpty(t, path)
+	assert.DirExists(t, path)
```

Comment:
So the test will create files there? Should we clean them up?

Reply:
nice catch